server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -275,6 +285,31 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
         Setting.Property.Deprecated
     );
 
+    /**
+     * Sets the cache size for snapshot repository data, defaulting to 1% of node heap memory. The maximum is 5%.
+     */
+    public static final Setting<ByteSizeValue> SNAPSHOT_REPOSITORY_DATA_CACHE_SIZE = new Setting<>(
+        SNAPSHOT_REPOSITORY_DATA_CACHE_SIZE_SETTING_NAME,
+        SNAPSHOT_REPOSITORY_DATA_CACHE_SIZE_DEFAULT_PERCENTAGE + "%",
+        (s) -> {
+            ByteSizeValue userDefinedLimit = parseBytesSizeValueOrHeapRatio(s, SNAPSHOT_REPOSITORY_DATA_CACHE_SIZE_SETTING_NAME);
```

Comment:
Hello @ashking94, thank you for your review.

I have added an integration test to `BlobStoreRepositoryTests`.

- `testSnapshotRepositoryDataCacheDefaultSetting`: This integration test checks if the field `repositoryDataCacheThreshold` in `BlobStoreRepository` is correctly assigned the value `min(max(500KB, 1% of heap), Integer.MAX)` when no value is provided by the user.

I wanted to add more integration tests (e.g., verifying that `Integer.MAX` is selected when the memory is assumed to be 250GB), but it was difficult to mock objects like `JvmInfo` or `static methods`, and generating enough data to test memory-related aspects was challenging. Therefore, I added unit tests instead:

- `testHeapThresholdUsed`: Verifies that `defaultThresholdOfHeap` is returned when it is greater than `defaultAbsoluteThreshold`.
- `testAbsoluteThresholdUsed`: Checks that `defaultAbsoluteThreshold` is returned when it is greater than `defaultThresholdOfHeap`.
- `testThresholdCappedAtIntMax`: Tests that the return value is limited to `Integer.MAX_VALUE` when `defaultThresholdOfHeap` exceeds `Integer.MAX_VALUE`.

Regarding `userDefinedLimit`, if the user doesn’t set any value, `calculateDefaultSnapshotRepositoryDataCacheThreshold()` will be executed. It is not directly related to `userDefinedLimit`. 😄

For parts where it was difficult to write test code, I performed E2E tests by building with Docker. Details related to the E2E tests have been added to the PR description.

Any further feedback is greatly appreciated. Thank you!

Reply:
I believed that a static setting, rather than dynamic, was more suitable here to avoid further amplifying the unpredictability of SoftReference.

If you have any further feedback, I would greatly appreciate it!